examples/hcp-aks-demo/main.tf

```diff
@@ -33,43 +34,54 @@ module "network" {
   nsg_ids = { for i, subnet in keys(var.vnet_subnets) : subnet => azurerm_network_security_group.nsg.id }
 
   depends_on = [azurerm_resource_group.rg]
+
 }
 
 # Create an HCP HVN.
-resource "hcp_hvn" "hvn" {
-  cidr_block     = var.hvn_cidr_block
-  cloud_provider = "azure"
-  hvn_id         = var.hvn_id
-  region         = var.hvn_region
+# resource "hcp_hvn" "hvn" {
+#   cidr_block     = var.hvn_cidr_block
+#   cloud_provider = "azure"
+#   hvn_id         = var.hvn_id
+#   region         = var.hvn_region
+# }
+
+data "hcp_hvn" "example" {
+  hvn_id = "hvn-azure-westus2"
 }
 
+# Note: Uncomment the below module to setup peering for connecting to a private HCP Consul cluster
 # Peer the HVN to the vnet.
-module "hcp_peering" {
-  source  = "hashicorp/hcp-consul/azurerm"
-  version = "~> 0.2.8"
+# module "hcp_peering" {
+#   source  = "hashicorp/hcp-consul/azurerm"
+#   version = "~> 0.2.8"
 
-  hvn    = hcp_hvn.hvn
-  prefix = var.cluster_id
+#   hvn    = hcp_hvn.hvn
+#   prefix = var.cluster_id
 
-  security_group_names = [azurerm_network_security_group.nsg.name]
-  subscription_id      = data.azurerm_subscription.current.subscription_id
-  tenant_id            = data.azurerm_subscription.current.tenant_id
+#   security_group_names = [azurerm_network_security_group.nsg.name]
+#   subscription_id      = data.azurerm_subscription.current.subscription_id
+#   tenant_id            = data.azurerm_subscription.current.tenant_id
 
-  subnet_ids = module.network.vnet_subnets
-  vnet_id    = module.network.vnet_id
-  vnet_rg    = azurerm_resource_group.rg.name
-}
+#   subnet_ids = module.network.vnet_subnets
+#   vnet_id    = module.network.vnet_id
+#   vnet_rg    = azurerm_resource_group.rg.name
+# }
 
 # Create the Consul cluster.
-resource "hcp_consul_cluster" "main" {
-  cluster_id      = var.cluster_id
-  hvn_id          = hcp_hvn.hvn.hvn_id
-  public_endpoint = true
-  tier            = var.tier
+# resource "hcp_consul_cluster" "main" {
+#   cluster_id         = var.cluster_id
+#   hvn_id             = hcp_hvn.hvn.hvn_id
+#   public_endpoint    = true
+#   tier               = var.tier
+#   min_consul_version = "v1.14.0"
```

Comment:
The default min version for demo examples would be 1.14.

Reply:
My only question, comment, concern rn is about this bit. This example file (despite the name) gets turned into the UI template: https://github.com/hashicorp/terraform-azurerm-hcp-consul/blob/main/hcp-ui-templates/aks/main.tf#L153

So I'd expect the frontend template to break for customers that use a default < 1.14.0 Consul cluster..? (Is that right/wrong? This is a gray area for me, eg idk if the helm:Consul versions are in lock-step)

A rough idea if that ^ is an issue is to have and create two AKS UI templates and the frontend selects between the two based on which version of Consul is being deployed.

Reply:
> So I'd expect the frontend template to break for customers that use a default < 1.14.0 Consul cluster..

AFAICT, there is no version selection allowed in the HCP UI quickstart templates. The version defaults to the recommended version (per the TF provider/consul-service API). 
The idea is once 1.14 is out, we want to have our example UI templates /TF configs use the Agentless model.

Reply:
Ahhh right, completely forgot, makes sense 👍